packages/flutter_tools/lib/src/build_runner/devfs_web.dart

```diff
@@ -139,16 +143,26 @@ class WebAssetServer implements AssetReader {
     // Attempt to look up the file by URI.
     if (_files.containsKey(requestPath)) {
       final List<int> bytes = getFile(requestPath);
+      final String etag = bytes.hashCode.toString();
```

Comment:
I don't think `List` actually has a useful `hashCode` by default - you should md5 hash the bytes or something like that (it is just falling back on Object.hashCode I think).

Reply:
Although maybe that is actually fine? If this is always getting the same exact list instance then it will work.

Reply:
I'll add a note here, this works because the buffer is only updated when the frontend_server gives new contents. It acts like a revision string.